### PR TITLE
Show image pull progress on dusty up

### PR DESCRIPTION
**Pull request by thieman**
@paetling

### CODE REVIEW

File: dusty/commands/utils.py [hunk 1]
```diff
@@ -20,3 +23,29 @@ def exec_docker(*args):
...
+    progress if it is attached to a TTY."""
+
+    updated_env = copy(os.environ)
+    updated_env.update(get_docker_env())
+    args += (updated_env,)
+    executable = args[0]
+
+    child_pid, pty_fd = pty.fork()
+    if child_pid == 0:
+        os.execle(_executable_path(executable), *args)
```
Comment (paetling) [inline dusty/commands/utils.py]: FORKING !

File: dusty/commands/utils.py [hunk 2]
```diff
@@ -20,3 +23,29 @@ def exec_docker(*args):
...
+    else:
+        child_process = psutil.Process(child_pid)
+        terminal = os.fdopen(pty_fd, 'r', 0)
+        with streaming_to_client():
+            while child_process.status() == 'running':
+                output = terminal.read(1)
+                log_to_client(output)
+        _, exit_code = os.waitpid(child_pid, 0)
+        if exit_code != 0:
+            raise subprocess.CalledProcessError(exit_code, ' '.join(args[:-1]))
```
Comment (paetling) [inline dusty/commands/utils.py]: SO COOL!

Comment (thieman) [inline dusty/commands/utils.py]: PRAISE SCIENCE

File: dusty/log.py [hunk 3]
```diff
@@ -57,3 +60,19 @@ def configure_client_logging():
...
+    receive many one-byte emissions. This context handler
+    was originally created for streaming Compose up's
+    terminal output through to the client and should only be
+    used for similarly complex circumstances."""
+    client_logger.propagate = False
+    handler = client_logger.handlers[0]
+    handler.append_newlines = False
+    yield
+    client_logger.propagate = True
+    handler.append_newlines = True
```
Comment (paetling) [inline dusty/log.py]: this should really restore the old state not just clear the whole state to true right?

Comment (thieman) [inline dusty/log.py]: Yeah I tried that and was seeing issues with it, lemme try again

File: dusty/commands/utils.py [hunk 4]
```diff
@@ -20,3 +23,29 @@ def exec_docker(*args):
     updated_env.update(get_docker_env())
     args += (updated_env,)
     os.execle(_executable_path('docker'), 'docker', *args)
+
+def pty_fork(*args):
```
Comment (paetling) [inline dusty/commands/utils.py]: wondering if this should go in the file with: check_and_log_output_and_error_demoted instead of this file.

Comment (thieman) [inline dusty/commands/utils.py]: It needs the Docker env stuff from the Docker module, if we put it higher we'll get circular imports

**Pull request merged**